messaging/kafka/src/main/java/io/helidon/messaging/kafka/SimpleKafkaConsumer.java

```diff
@@ -0,0 +1,322 @@
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import io.helidon.common.context.Context;
+import io.helidon.common.context.Contexts;
+import io.helidon.config.Config;
+import io.helidon.messaging.kafka.connector.KafkaMessage;
+import io.helidon.messaging.kafka.connector.SimplePublisher;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.errors.WakeupException;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+/**
+ * Simple Kafka consumer covering basic use-cases.
+ * Configurable by Helidon {@link io.helidon.config.Config Config},
+ * For more info about configuration see {@link KafkaConfigProperties}
+ * <p>
+ * Usage:
+ * <pre>{@code
+ *   try (SimpleKafkaConsumer<Long, String> c = new SimpleKafkaConsumer<>("test-channel", Config.create())) {
+ *         c.consumeAsync(r -> System.out.println(r.value()));
+ *   }
+ * }</pre>
+ *
+ * @param <K> Key type
+ * @param <V> Value type
+ * @see KafkaConfigProperties
+ * @see io.helidon.config.Config
+ */
+public class SimpleKafkaConsumer<K, V> implements Closeable {
+
+    private static final Logger LOGGER = Logger.getLogger(SimpleKafkaConsumer.class.getName());
+    private final KafkaConfigProperties properties;
+
+    private final AtomicBoolean closed = new AtomicBoolean(false);
+    private final PartitionsAssignedLatch partitionsAssignedLatch = new PartitionsAssignedLatch();
+    private final String consumerId;
+    private ExecutorService executorService;
+    private ExecutorService externalExecutorService;
+    private final List<String> topicNameList;
+    private final KafkaConsumer<K, V> consumer;
+
+    /**
+     * Kafka consumer created from {@link io.helidon.config.Config config}
+     * see configuration {@link KafkaConfigProperties example}.
+     *
+     * @param channelName key in configuration
+     * @param config      Helidon {@link io.helidon.config.Config config}
+     * @see KafkaConfigProperties
+     * @see io.helidon.config.Config
+     */
+    public SimpleKafkaConsumer(String channelName, Config config) {
```

Comment:
Never use public constructors in Helidon.
If this class should be public (in which case the name should change), use a `Builder` pattern as in other Helidon classes.
You may have a static factory method, such as `create(String, Config)`, but should be limited to one such method.

Reply:
I think that class was designed with 2 purposes:
- To be integrated with cdi
- To be used from any other code to consume from Kafka.

I have simplified this class for the first point. There is only one constructor.